crates/noirc_evaluator/src/ssa/optim.rs

```diff
@@ -116,6 +152,69 @@ pub fn find_similar_instruction(
     None
 }
 
+pub enum CseAction {
+    Replace, //replace the instruction
+    Remove,  //remove the instruction
+    Keep,    //keep the instruction
+}
+
+pub fn find_similar_mem_instruction(
+    eval: &IRGenerator,
+    op: node::Operation,
+    ins_id: arena::Index,
+    lhs: arena::Index,
+    rhs: arena::Index,
+    anchor: &HashMap<node::Operation, VecDeque<arena::Index>>,
+) -> (arena::Index, CseAction) {
+    match op {
+        node::Operation::Load(_) => {
+            for iter in anchor[&op].iter().rev() {
+                if let Some(ins_iter) = eval.try_get_instruction(*iter) {
+                    match ins_iter.operator {
+                        node::Operation::Load(_) => {
+                            if ins_iter.lhs == lhs {
+                                return (*iter, CseAction::Replace);
+                            }
+                        }
+                        node::Operation::Store(_) => {
+                            if ins_iter.rhs == lhs {
+                                return (ins_iter.lhs, CseAction::Replace);
```

Comment:
Replace with what?

Reply:
What does the returned `arena::Index` here signify? From my understanding, we replace `op` with it in the Replace case, what about the Remove and Keep cases?

Reply:
Did you see the added comment on the function? 
If no, does it answer your question?
(and if yes that means I am not clear :) )